booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -297,10 +297,17 @@ private static Consumer<MultiAttributeValidator> allDg(Consumer<MultiAttributeVa
     ALL_SHIPMENT_CUTOFF_TIMES,
     JsonAttribute.unique("cutOffDateTimeCode")
   );
-  private static final JsonContentCheck AMF_CC_MTC_COMBINATION_VALIDATIONS = JsonAttribute.allIndividualMatchesMustBeValid(
-    "Validate combination of 'countryCode' and 'manifestTypeCode' in 'advanceManifestFilings'",
-    ALL_AMF,
-    JsonAttribute.combineAndValidateAgainstDataset(BookingDataSets.AMF_CC_MTC_COMBINATIONS, "countryCode", "manifestTypeCode")
+
+  private static final Predicate<JsonNode> HAS_NON_NULL_AMF_ITEM = (reqEquipNode) -> reqEquipNode.hasNonNull("countryCode")
+    && reqEquipNode.hasNonNull("manifestTypeCode");
+
+  private static final JsonRebaseableContentCheck AMF_CC_MTC_COMBINATION_VALIDATIONS = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All utilizedTransportEquipments with a reefer ISO Equipment Code must have at least isNonOperatingReefer",
+    (mav) -> mav.submitAllMatching("advanceManifestFilings.*"),
+    JsonAttribute.ifMatchedThen(
+      HAS_NON_NULL_AMF_ITEM,
+      JsonAttribute.combineAndValidateAgainstDataset(BookingDataSets.AMF_CC_MTC_COMBINATIONS, "countryCode", "manifestTypeCode")
+    )
   );
```

Comment:
To be honest, I do not understand why this logic is necessary.

The `submitAllMatching` on an array would should only match existing array elements and we should not have null elements.

Please show me the payload flagged as incorrect when you do not have this `HAS_NON_NULL_AMF_ITEM` guard in it.